### PR TITLE
fix: add try/throw within util/regex

### PR DESCRIPTION
**Pull request by rarkins**

### CODE REVIEW

File: test/util/regex.spec.ts [hunk 1]
```diff
@@ -1,13 +1,16 @@
...
+import { regEx } from '../../lib/util/regex';
 
 describe('util/regex', () => {
   beforeEach(() => {
     jest.resetModules();
   });
 
   it('uses RE2', () => {
-    expect(RegEx).toEqual(RE2);
+    expect(regEx('foo') instanceof RE2).toBe(true);
```
Comment (ViceIce) [inline test/util/regex.spec.ts]:
```suggestion
    expect(regEx('foo')).toBeInstanceOf(RE2);
```

File: test/util/regex.spec.ts [hunk 2]
```diff
@@ -16,6 +19,6 @@ describe('util/regex', () => {
     });
 
     const regex = require('../../lib/util/regex');
-    expect(regex.RegEx).toEqual(RegExp);
+    expect(regex.regEx('foo') instanceof RegExp).toBe(true);
```
Comment (ViceIce) [inline test/util/regex.spec.ts]:
```suggestion
    expect(regex.regEx('foo')).toBeInstanceOf(RegExp);
```

**Review by viceice — CHANGES_REQUESTED** [review]

**Comment by ViceIce** [comment]
`lib/versioning/regex/index`: unused logger causes lint failure

**Review by viceice — APPROVED** [review]
LGTM

**Pull request merged**